src/StatsdClient/BufferBuilder.cs

```diff
@@ -0,0 +1,45 @@
+using System.Text;
+
+namespace StatsdClient
+{
+    /// <summary>
+    /// Append string values to a fixed size bytes buffer.
+    /// </summary>
+    class BufferBuilder
+    {
+        readonly IBufferBuilderHandler _handler;
+        readonly byte[] _buffer;
+
+        public BufferBuilder(IBufferBuilderHandler handler, int bufferCapacity)
+        {
+            _buffer = new byte[bufferCapacity];
+            _handler = handler;
+        }
+
+        public int Length { get; private set; }
+        public int Capacity { get { return _buffer.Length; } }
+
+        public bool Add(string value)
+        {
+            var byteCount = Encoding.UTF8.GetByteCount(value);
+            if (byteCount > _buffer.Length)
+                return false;
```

Comment:
On the .NET Tracer team (see https://github.com/DataDog/dd-trace-dotnet) we use a StyleCop configuration file that won't allow us to compile code that doesn't pass. I suggest you consider this for this repo. I'll add a separate comment on what that file is in our repo. In the above case, use braces for `if` statements.

Reply:
The stylecop file is [stylecop.json](https://github.com/DataDog/dd-trace-dotnet/blob/master/stylecop.json). Some [documentation](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/documentation/Configuration.md) on it.

Reply:
Thank you for providing stylecop config file. I do agree having a consistent code style in an open source project is important. 
At the same time, this PR is already quite big and another PR is waiting on this one to be merged so I think it is not the best timing.

Reply:
ok, just wanted to provide some feedback to show what we've done on the .net tracer team.

Reply:
Thank you @bobuva - how do you feel about us following up with all the corresponding style (non-logical) changes in a follow-up PR? We will 100% add stylecop to the repo, and take that step towards aligning with APM. 🙇

Reply:
That's fine -- thanks!